manual/trees/index.md

```diff
@@ -511,42 +511,38 @@ The scatterplot is drawn.
 Note that not each `(x,y) point on a scatterplot represents two values in your N−tuple. In fact, the scatterplot is a grid and each square in
 the grid is randomly populated with a density of dots that’s proportional to the number of values in that grid.
 
-### Indexing trees
-
-- Use [TTree::BuildIndex()](https://root.cern/doc/master/classTTree.html#a3f6b5bb591ff7a5bd0b06eea6c12b998){:target="_blank"} method to build an index table using expressions depending on the value in the leaves.
-
-The index is built in the following way:
-- A pass on all entries is made like in [TTree::Draw()](https://root.cern/doc/master/classTTree.html#a3f6b5bb591ff7a5bTTree::Draw()d0b06eea6c12b998){:target="_blank"}.
-- `var1` = `majorname`
-- `var2` = `minorname`
-- `sel = 231` × _majorname_ + _minorname_
-- For each entry in the tree the `sel` expression is evaluated and the result array is sorted into `fIndexValues`.
-
-Once the index is calculated, an entry can be retrieved with
+### Indexing a tree
+
+- Use [TTree::BuildIndex()](https://root.cern/doc/master/classTTree.html#a3f6b5bb591ff7a5bd0b06eea6c12b998){:target="_blank"} to build an index table over expressions that depend on the value in the leaves.
+This index is similar to database indexes:
+it allows to quickly determine the tree entry number corresponding to the value of an expression.
+These expressions should be both equality comparable (that is, not use floating point numbers where precision might cause the index lookup to fail) and unique, to make sure you get the tree entry you expect.
+For high-energy physics, a common example could be a combination of run number and event number:
+while each one of them might have duplications, their combination is guaranteed to be unique. 
+
+To build an index, define a major and optionally a minor expression, for instance above `Run` and `Event`.
```

Comment:
Says above but the example is below?

Reply:
Should we say they are leaves?

Reply:
No, they are really expressions (eg. `run%1000`, `event_part_1 * 1000 + event_part_2`)